mllib/src/test/scala/org/apache/spark/ml/util/DefaultReadWriteTest.scala

```diff
@@ -107,24 +133,34 @@ trait DefaultReadWriteTest extends TempDirectory { self: Suite =>
     }
     val model = estimator.fit(dataset)
 
-    // Test Estimator save/load
-    val estimator2 = testDefaultReadWrite(estimator)
-    testParams.foreach { case (p, v) =>
-      val param = estimator.getParam(p)
-      assert(estimator.get(param).get === estimator2.get(param).get)
+    val testFunctions = if (testParams.contains("initialModel")) {
+      Map(("initialModel", checkModelData.asInstanceOf[(Any, Any) => Unit]))
```

Comment:
Won't model equality depend on algorithm specific details? Each model has different parameters and we really need to use those to check equality...